met-api/src/met_api/services/email_verification_service.py

```diff
@@ -53,11 +62,30 @@ def verify(self, verification_token, survey_id):
         return db_email_verification
 
     @staticmethod
-    def send_verification_email(email_verification: EmailVerificationSchema) -> DefaultMethodResult:
-        """Send an verification email."""
-        print(email_verification)
-        # TODO: send email.
-        return DefaultMethodResult(True, '', 0)
+    def _send_verification_email(email_verification: dict) -> None:
+        """Send an verification email.Throws error if fails."""
+        sender = current_app.config.get('MAIL_FROM_ID')
+        survey_id = email_verification.get('survey_id')
+        email_to = email_verification.get('email_address')
+        survey: SurveyModel = SurveyModel.get_survey(survey_id)
+        if not survey:
+            raise ValueError('Survey not found')
+        template = ENV.get_template('email_templates/email_verification.html')
+        # TODO make it read from config
+        subject = 'survey link - link expires in 24h'
+        app_url = f"{g.get('origin_url', '')}/" \
+                  f"{current_app.config.get('SURVEY_PATH')}" \
+                  f"{survey_id}/{email_verification.get('verification_token')}"
+        body = template.render(engagement_name=survey.get('engagement').get('name'), url=app_url)
```

Comment:
what if engagement was None?

Reply:
i added a check.its a private method ;ideally shud be just receiving what it needs ;but it was lot of refactoring ..will try to improve it..